src/mmw/js/src/modeling/gwlfeModificationConfig.js

```diff
@@ -0,0 +1,286 @@
+            [outputName, fractionVal * 100]
+        ]);
+    }
+
+    return {
+        dataModelNames: [n23Name],
+        userInputNames: [areaToConvertName],
+        validate: makeThresholdValidateFn(n23Name, AREA, areaToConvertName),
+        computeOutput: makeComputeOutputFn(n23Name, areaToConvertName, getOutput)
+    };
+}
+
+function makeAeuBmpConfig(outputName) {
+    return {
+        dataModelNames: [],
+        userInputNames: [percentAeuToConvertName],
+        validate: makePercentValidateFn(percentAeuToConvertName),
+        computeOutput: function(dataModel, cleanUserInput) {
+            var percentAeuToConvert = cleanUserInput[percentAeuToConvertName],
+                output = {},
+                infoMessages = {};
+
+            if (percentAeuToConvert) {
+                output[outputName] = percentAeuToConvert;
+            }
+
+            return formatOutput(infoMessages, output);
+        }
+
+    };
+}
+
+function makeRuralStreamsBmpConfig(outputName) {
+    function getOutput(inputVal) {
+        return toObject([
+            [outputName, inputVal]
+        ]);
+    }
+
+    // Should the fraction be calculated against n42b, the total stream length?
+    // or n42, the length of streams in ag areas?
+    return {
+        dataModelNames: [n42Name, n42bName],
+        userInputNames: [lengthToConvertName],
+        validate: makeThresholdValidateFn(n42bName, LENGTH, lengthToConvertName),
+        computeOutput: makeComputeOutputFn(n42bName, lengthToConvertName, getOutput)
+    };
+}
+
+function makeUrbanStreamsBmpConfig(getOutput) {
+    return {
+        dataModelNames: [UrbLengthName],
+        userInputNames: [lengthToConvertName],
+        validate: makeThresholdValidateFn(UrbLengthName, LENGTH, lengthToConvertName),
+        computeOutput: makeComputeOutputFn(UrbLengthName, lengthToConvertName, getOutput)
+    };
+}
+
+function makeUrbanAreaBmpConfig(getOutput) {
+    return {
+        dataModelNames: [UrbAreaTotalName],
+        userInputNames: [areaToConvertName],
+        validate: makeThresholdValidateFn(UrbAreaTotalName, AREA, areaToConvertName),
+        computeOutput: makeComputeOutputFn(UrbAreaTotalName, areaToConvertName, getOutput)
+    };
+}
+
+/*
+    configs contains a mapping from the BMP key to information needed
+    to generate the UI for that modification.
+    - dataModelNames is a list of variable names in the Mapshed data model
+    (received from the backend) which should be displayed to the user.
+    - userInputNames is a list of non-Mapshed variables used to reference values
+    the user enters into the UI. currently all the BMPs only reference a single
+    user input variable, so using a list is overkill, but it could be useful in
+    the future.
+    - validate is a function which validates the user input using values from
+    the data model.
+    - computeOutput is a function which computes the output of the UI for a BMP,
+    the values for a set of Mapshed variables.
+*/
```

Comment:
Nice comment. I think on future projects we should consider using something like [JSDoc](http://usejsdoc.org/about-getting-started.html)